src/core/credGrainView.js

```diff
@@ -54,77 +54,137 @@ export type ParticipantsOptions = {|
 export class CredGrainView {
   _credGraph: CredGraph;
   _ledger: Ledger;
-  _defaultTimeScope: TimeScopedCredGrainView;
+  _participants: $ReadOnlyArray<ParticipantCredGrain>;
+  _intervals: IntervalSequence;
 
   constructor(credGraph: CredGraph, ledger: Ledger) {
     this._credGraph = credGraph;
     this._ledger = ledger;
-    this._defaultTimeScope = new TimeScopedCredGrainView(
-      credGraph,
-      ledger,
-      -Infinity,
-      Infinity
+    this._intervals = deepFreeze(credGraph.intervals());
+
+    const graphParticipants = new Map<IdentityId, GraphParticipant>();
+    for (const participant of credGraph.participants()) {
+      graphParticipants.set(participant.id, participant);
+    }
+
+    this._participants = deepFreeze(
```

Comment:
hmmm. this technically mutates the input, since it deepFreezes otherwise-mutable objects like `graphParticipant.credPerInterval`. I think it's fine because those are read-only anyway, but just flagging it.